[DOCS-1422][DOCS-1375] Document workspace templates (#1325)
wandb/docs: content/guides/models/track/workspaces.md

### Share a workspace view
Share your customized workspace with your team by sharing the workspace URL directly. All users with access to the workspace project can see the saved Views of that workspace.

## Workspace templates
Use _workspace templates_ to quickly create new workspaces using an existing workspace's settings to new workspaces. Currently, a workspace template can include line plot settings.

### Default workspace settings
After you configure a workspace template, you can't delete it or stop using it. Instead, you can update it to use the default settings:

| Setting | Default |
|-------|----------
| X axis             | Step |
| Smoothing type     | Time weight EMA |
| Smoothing weight   | 0 |
| Max runs           | 10 |
| Grouping in charts | on |
| Group aggregation  | Mean |

### Create a workspace template

> 1. Open or create a workspace that has the [line plot settings]({{< relref "https://docs.wandb.ai/guides/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}) you want to templatize.

Author: ```suggestion
1. Open or create a workspace that has the [line plot settings]({{< relref "/guides/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}) you want to templatize.
```

1. Save the settings as a workspace template.

  1. Click the workspace's `...` action menu at the top of the page near the **Undo** and **Redo** arrow icons.
  1. Click **Save personal workspace template**.
  1. Review the line plot settings that the template will use, then click **Save**.

### Update a workspace template
After you create a workspace template, follow these steps to update it:

> 1. Open any workspace that uses the existing template.

Reviewer: you can really just go to or open any workspace... you don't need to open one that uses and existing template

Author: This is confusing. How can I know if a workspace is currently using the template or not?

Reviewer: hmmm why do you think that is important to the user?

you can change the settings or the workspace at anytime... this is really just trying to give you a better default experience.

Author: I think I got my mental map straightened out in the interim. Is there more I need to correct?

1. Modify the workspace's line plot settings. For example, set the number of runs to include to `11`.
1. To save the changes to the template, click the workspace's `...` action menu at the top of the page near the **Undo** and **Redo** arrow icons, then click **Update personal workspace template**.
    {{% alert %}}
    If you see **Save personal workspace template**, no workspace template has been created yet. Refer to [Create a workspace template]({{< relref "#create-a-workspace-template" >}}).
    {{% /alert %}}
1. Verify the new line plot settings, then click **Update**. The template is updated and reapplied to all workspaces that use it.

> 

Reviewer: these all use `1.`

Author: Yep, if you look at the HTML preview you will see that it auto-numbers them correctly.

### Share a workspace view
Share your customized workspace with your team by sharing the workspace URL directly. All users with access to the workspace project can see the saved Views of that workspace.

## Workspace templates

<!-- Tracked in https://wandb.atlassian.net/browse/DOCS-1543 -->
{{% alert %}}
{{< readfile file="/_includes/workspace-templates-private-preview-feature.md" >}}
{{% /alert %}}

> Use _workspace templates_ to quickly create workspaces using the same settings as an existing workspace instead of the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}). Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).

Author: ```suggestion
Use _workspace templates_ to automatically apply custom settings from an existing workspace to new workspaces without updating each [default setting]({{< relref "#default-workspace-settings" >}}) individually. Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).
```

### Default workspace settings

> By default, new workspaces use these default settings for line plots:

Author: ```suggestion
By default, new workspaces use these default settings:

#### Line plots

```
Here's a compromise. :)

### Share a workspace view
Share your customized workspace with your team by sharing the workspace URL directly. All users with access to the workspace project can see the saved Views of that workspace.

> ## Workspace templates

Reviewer 2: We likely need a callout letting people know this is only available for Enterprise Edition customers.

Author: Added this to the same admonition where the private preview detail is shown. When we remove that, I'll shift this to the normal enterprise-only admonition.

Author: Updated this after Slack conversation to note that it is Enterprise-only for everyone, and private preview for Dedicated Cloud and Self-Hosted.

<!-- Tracked in https://wandb.atlassian.net/browse/DOCS-1543 -->
{{% alert %}}
{{< readfile file="/_includes/private-preview-feature.md" >}}
This feature requires an [Enterprise](https://wandb.ai/site/pricing/) license.
{{% /alert %}}

> Use _workspace templates_ to quickly create workspaces using the same settings as an existing workspace instead of the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}). Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).

Reviewer 3: ```suggestion
Use _workspace templates_ to create workspaces using the same settings as an existing workspace instead of the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}). Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).
```

Reviewer 3: "Default workspace settings" make it sound like there are multiple settings that can be configured (e.g. layout of the workspace, how many panels show up, etc.). 

Let's be more specific and mentions that it only (or seems to only)  apply to line plot panels.

Author: @ktmathews89 @tristan-spaulding2 WDYT? I left Katie's language in place to let the reader know that more settings are coming. If I do the change Noah is suggesting, I worry that the reader will think that the only thing we ever plan to include is line plot settings.

Reviewer 3: With the update, keep in mind not dating the docs: https://developers.google.com/style/timeless-documentation

Reviewer: it's going to be expanded for additional settings... so I personally would keep it generalized.

| Smoothing type     | Time weight EMA |
| Smoothing weight   | 0 |
| Max runs           | 10 |
| Grouping in charts | on |
| Group aggregation  | Mean |

### Configure your workspace template
1. Open any workspace or create a new one.
1. Configure the workspace's [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}) according to your preferences.
1. Save the settings to your workspace template:
    1. Click the workspace's `...` action menu at the top of the page near the **Undo** and **Redo** arrow icons.
    1. Click **Save personal workspace template**.
    1. Review the line plot settings for the template, then click **Save**.

New workspaces will use these settings instead of the defaults.

{{% alert %}}
To stop using a customized workspace template, you can [update your workspace template]({{< relref "#update-a-workspace-template" >}}) back to the [default setting]({{< relref "#default-workspace-settings" >}}).

> {{% /alert %}}

Reviewer 2: We probably can delete this since you also add this warning at the end, right?

    1. Review the line plot settings for the template, then click **Save**.

New workspaces will use these settings instead of the defaults.

### Update your workspace template
Follow these steps to update your workspace template:

1. Open any workspace.
1. Modify the workspace's line plot settings. For example, set the number of runs to include to `11`.

> 1. To save the changes to the template, click the workspace's `...` action menu at the top of the page near the **Undo** and **Redo** arrow icons, then click **Update personal workspace template**.

Reviewer 3: Do users know that "..." is known as an action menu? If not, should we describe what it looks like instead?

Author: Maybe the reader of _this_ page doesn't know that. In the panels pages, they do know. I'll adjust this a bit.

>     {{% /alert %}}

Reviewer 2: This felt a bit awkward, do we need it?

Author: I'll take it out. I was worried about confusion if the user doesn't see the exact text, but perhaps they will be able to figure it out.

### Share a workspace view
Share your customized workspace with your team by sharing the workspace URL directly. All users with access to the workspace project can see the saved Views of that workspace.

## Workspace templates

<!-- Tracked in https://wandb.atlassian.net/browse/DOCS-1543 -->
{{% alert %}}
{{< readfile file="/_includes/workspace-templates-private-preview-feature.md" >}}
{{% /alert %}}

Use _workspace templates_ to automatically apply custom settings from an existing workspace to new workspaces without updating each [default setting]({{< relref "#default-workspace-settings" >}}) individually. Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).

### Default workspace settings
By default, new workspaces use these default settings:

#### Line plots

| Setting | Default |
|-------|----------
| X axis             | Step |

> | Smoothing type     | Time weight EMA |

Reviewer 3: Clarify that EMA == exponential moving average
```suggestion
| Smoothing type     | Time weight exponential moving average (EMA) |
```

| Smoothing weight   | 0 |
| Max runs           | 10 |
| Grouping in charts | on |
| Group aggregation  | Mean |

### Configure your workspace template
1. Open any workspace or create a new one.
1. Configure the workspace's [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}) according to your preferences.
1. Save the settings to your workspace template:
    1. At the top of the workspace, click the action menu `...` near the **Undo** and **Redo** arrow icons.
    1. Click **Save personal workspace template**.
    1. Review the line plot settings for the template, then click **Save**.

> New workspaces will use these settings instead of the defaults.

Author: ```suggestion
New workspaces will use these settings instead of the defaults.

If you decide to stop using a customized workspace template, you can [update it]({{< relref "#update-your-workspace-template" >}}) to use the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}).
```

### Update your workspace template
Follow these steps to update your workspace template:

1. Open any workspace.
1. Modify the workspace's line plot settings. For example, set the number of runs to include to `11`.
1. To save the changes to the template, click the action menu `...` near the **Undo** and **Redo** arrow icons, then click **Update personal workspace template**.
1. Verify the new line plot settings, then click **Update**. The template is updated, and reapplied to all workspaces that use it.

> {{% alert %}}

Reviewer 3: I wonder if it makes sense to have this info also (or instead) in the previous section? Just a thought.

Author: Done, also demoted this one from being an alert.

> {{% alert %}}
> If you decide to stop using a customized workspace template, you can [update it]({{< relref "#update-your-workspace-template" >}}) to use the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}).
> {{% /alert %}}

Author: ```suggestion
If you decide to stop using a customized workspace template, you can [update it]({{< relref "#update-your-workspace-template" >}}) to use the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}).
```
